src/__tests__/reactUtils-test.js

```diff
@@ -0,0 +1,232 @@
+      sequence: [
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS * 2, expectedOutput: false },
+      ],
+    },
+    {
+      initialValue: false,
+      sequence: [
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: false },
+      ],
+    },
+    {
+      initialValue: true,
+      sequence: [{ waitBefore: MS / 2, expectedOutput: false }],
+    },
+    {
+      initialValue: true,
+      sequence: [{ waitBefore: MS * 2, expectedOutput: true }],
+    },
+    {
+      initialValue: true,
+      sequence: [
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS, expectedOutput: false },
+      ],
+    },
+    {
+      initialValue: true,
+      sequence: [
+        { waitBefore: MS * 2, expectedOutput: true, thenUpdateInputTo: false },
+        { waitBefore: MS / 2, expectedOutput: false },
+        { waitBefore: MS, expectedOutput: false },
+      ],
+    },
+    {
+      initialValue: true,
+      sequence: [
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 2, expectedOutput: false, thenUpdateInputTo: false },
+        { waitBefore: MS * 2, expectedOutput: false },
+      ],
+    },
+    {
+      initialValue: true,
+      sequence: [
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 5, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 5 - 1, expectedOutput: false, thenUpdateInputTo: true },
+        { waitBefore: MS / 5 + 1, expectedOutput: true },
+      ],
+    },
+  ];
+
+  for (let i = 0; i < sequencesToTest.length; i++) {
+    const currentSequence = sequencesToTest[i];
+    test(JSON.stringify(currentSequence, null, 2), async () => {
```

Comment:
These test "names" get pretty long in the Jest output. Let's add something like a `description` property to each of these test-case objects, and use that here.

Another benefit of that is that it'd provide a good place for a few words explaining the scenario each of these is meant to cover.

For example:
```js
    {
      description: "start false, wait short time",
      initialValue: true,
      sequence: [{ waitBefore: MS / 2, expectedOutput: false }],
    },
    {
      description: "start false, wait long time",
      initialValue: true,
      sequence: [{ waitBefore: MS * 2, expectedOutput: true }],
    },
```